src/components/pages/AllProjects.tsx

```diff
@@ -171,21 +190,27 @@ const AllProjects = () => {
                 defaultPageSize={25}
               />
             ) : (
-              <GenericTableWithData<
-                GetOrganizationsQuery,
-                GetOrganizationsQueryVariables,
-                OrganizationType
-              >
-                queryDocument={GetOrganizationsDocument}
-                columns={ORGANIZATION_COLUMNS}
-                rowLinkTo={organizationRowLink}
-                noData='No organizations'
-                pagePath='organizations'
-                showFilters
-                recordType='Organization'
-                defaultPageSize={25}
-                queryVariables={{}}
-              />
+              <Box>
```

Comment:
This extra `<Box>` was a quick and dirty solution to a bug I'd like to discuss more, both to improve my own React knowledge, and because I suspect there is a cleaner way to solve it. Here's the issue:

React seems to treat both `<GenericTableWithData>` components as the same component, regardless of `viewMode`, so it doesn't re-render when switching from Project to Organization. The `defaultFilters` param may change when `viewMode` changes, but because it is not used directly, but instead is passed in as the default value for the `filterValues` state [here](https://github.com/greenriver/hmis-frontend/blob/efbcb35416ec43fc23767d900980fea6c5ed8307/src/modules/dataFetching/components/GenericTableWithData.tsx#L178), changing the default value after the component has been rendered doesn't cause the `filterValues` state to change.

You can reproduce the issue locally by commenting out the `<Box>` wrapper on line 193 and 213. You should see an error when switching from Projects to Organizations: "Variable $filters of type OrganizationFilterOptions was provided invalid value for status (Field is not defined on OrganizationFilterOptions)". Inspecting in the chrome console, you can see that `filterValues` state in `GenericTableWithData` still includes the `status` attribute.

Anyway, adding a `Box` here basically tricks React, I guess, because if the dom tree is different for the 2 different modes then React will do a full re-render and treat them as separate components.

Reply:
Huh, thats a weird one. The issue goes away if you add `key='projectTable'` and `key='organizationTable'` to the two tables, that must help React figure out that they are different components. I'm not sure why it couldn't figure it out, though.

Reply:
Ah thank you, that is much cleaner 👍